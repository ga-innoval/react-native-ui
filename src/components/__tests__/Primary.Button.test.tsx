import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react-native';
import { PrimaryButton } from '../Primary.Button';
const eventData = {
  nativeEvent: {
    pageX: 20,
    pageY: 30,
  },
};

describe('<PrimaryButton loading={false} />', () => {
  const mockFn = jest.fn();

  beforeEach(() => {
    render(<PrimaryButton text="Press me" loading={false} onPress={mockFn} />);
  });

  it('should display text label', () => {
    expect(screen.getByText(/press me/i)).toBeDefined();
  });

  it('should call function on press', () => {
    const button = screen.getByText(/press me/i);
    fireEvent.press(button, eventData);
    expect(mockFn).toHaveBeenCalled();
  });
});

describe('<PrimaryButton loading={true} />', () => {
  const mockFn = jest.fn();

  beforeEach(() => {
    render(
      <PrimaryButton
        text="Press me"
        loading={true}
        onPress={mockFn}
        accessibilityHint="button"
      />
    );
  });

  it("shouldn't display text label", () => {
    expect(screen.queryByText(/press me/i)).toBeNull();
  });

  it("shouldn't call function on press", () => {
    const button = screen.getByAccessibilityHint('button');
    fireEvent.press(button, eventData);
    expect(mockFn).not.toHaveBeenCalled();
  });
});
