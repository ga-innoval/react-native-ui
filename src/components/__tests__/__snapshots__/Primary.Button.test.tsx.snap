// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrimaryButton custom style /> Should render correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "borderRadius": 10,
        "justifyContent": "center",
        "padding": 15,
      },
      {
        "marginTop": 10,
      },
      {
        "backgroundColor": "#232222",
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#7A7878",
          "fontSize": 16,
        },
        {
          "color": "white",
          "fontWeight": "bold",
        },
      ]
    }
  >
    Press me
  </Text>
</View>
`;

exports[`<PrimaryButton loading={false} /> should render correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "borderRadius": 10,
        "justifyContent": "center",
        "padding": 15,
      },
      null,
      {
        "backgroundColor": "#232222",
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#7A7878",
          "fontSize": 16,
        },
        {
          "color": "white",
          "fontWeight": "bold",
        },
      ]
    }
  >
    Press me
  </Text>
</View>
`;

exports[`<PrimaryButton loading={true} /> should render correctly 1`] = `
<View
  accessibilityHint="button"
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "borderRadius": 10,
        "justifyContent": "center",
        "padding": 15,
      },
      null,
      {
        "backgroundColor": "#232222",
      },
    ]
  }
>
  <ActivityIndicator
    color="white"
    size="small"
  />
</View>
`;
